src/main/java/com/hierynomus/msdtyp/FileTime.java

```diff
@@ -0,0 +1,59 @@
+package com.hierynomus.msdtyp;
+
+import java.util.Date;
+
+import static com.hierynomus.msdtyp.MsDataTypes.NANO100_TO_MILLI;
+import static com.hierynomus.msdtyp.MsDataTypes.WINDOWS_TO_UNIX_EPOCH;
+
+public class FileTime
+{
+    private final long windowsTimeStamp;
+
+    public static FileTime fromDate(Date date) {
+        return new FileTime(date.getTime() * NANO100_TO_MILLI + WINDOWS_TO_UNIX_EPOCH);
+    }
+
+    public static FileTime now() {
+        return ofEpochMillis(System.currentTimeMillis());
+    }
+
+    public static FileTime ofEpochMillis(long epochMillis) {
+        return new FileTime(epochMillis * NANO100_TO_MILLI + WINDOWS_TO_UNIX_EPOCH);
+    }
+
+    public FileTime(long windowsTimeStamp)
```

Comment:
Can you ensure it's consistently formatted? I.e. some methods have open curly at end of line (in line with current code style) and some have it on a new line 😉